filesystem/iso9660.ksy

```diff
@@ -5,269 +5,589 @@ meta:
+      - id: optional_occurrence_of_type_m_path_table
+        doc-ref: ecma-119 8.5.11
+        type: u4le
+      - id: directory_record_for_root_directory
+        doc-ref: ecma-119 8.5.12
+        type: directory_record_head
+        size: 0x22
+      - id: volume_set_identifier
+        doc-ref: ecma-119 8.5.13
+        type: volume_text_128
+      - id: publisher_identifier
+        doc-ref: ecma-119 8.5.14
+        type: volume_text_128
+      - id: data_preparer_identifier
+        doc-ref: ecma-119 8.5.15
+        type: volume_text_128
+      - id: application_identifier
+        doc-ref: ecma-119 8.5.16
+        type: volume_text_128
+      - id: copyright_file_identifier
+        doc-ref: ecma-119 8.5.17
+        type: volume_text_37
+      - id: abstract_file_identifier
+        doc-ref: ecma-119 8.5.18
+        type: volume_text_37
+      - id: bibliographic_file_identifier
+        doc-ref: ecma-119 8.5.19
+        type: volume_text_37
+      - id: volume_creation_date_and_time
+        doc-ref: ecma-119 8.5
+        type: datetime_long
+      - id: volume_modification_date_and_time
+        doc-ref: ecma-119 8.5
+        type: datetime_long
+      - id: volume_expiration_date_and_time
+        doc-ref: ecma-119 8.5
+        type: datetime_long
+      - id: volume_effective_date_and_time
+        doc-ref: ecma-119 8.5
+        type: datetime_long
+      - id: file_structure_version
+        doc-ref: ecma-119 8.5
+        type: s1
+#  volume_partition:
+#    seq:
+#      - id: todo
+#        type: u1
+  datetime_short:
+    doc-ref: ecma-119 9.1.5
+    seq:
+      - id: year
         type: u1
-      - id: file_unit_size
+      - id: month
         type: u1
-      - id: interleave_gap_size
+      - id: day
         type: u1
-      - id: vol_seq_num
-        type: u2bi
-      - id: len_file_name
+      - id: hour
         type: u1
-      - id: file_name
-        type: str
-        encoding: UTF-8
-        size: len_file_name
-      - id: padding
+      - id: min
         type: u1
-        if: len_file_name % 2 == 0
-      - id: rest
-        size-eos: true
-    instances:
-      extent_as_dir:
-        io: _root._io
-        pos: lba_extent.le * _root.sector_size
-        size: size_extent.le
-        type: dir_entries
-        if: file_flags & 2 != 0
-      extent_as_file:
-        io: _root._io
-        pos: lba_extent.le * _root.sector_size
-        size: size_extent.le
-        if: file_flags & 2 == 0
-  ## AKA "Path Table Entry"
-  path_table_le:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#The_Path_Table'
-    seq:
-      - id: entries
-        type: path_table_entry_le
-        repeat: eos
-  path_table_entry_le:
+      - id: sec
+        type: u1
+      - id: offset
+        type: s1
+  rrip_nm:
+    doc-ref: rrip 4.1.4
     seq:
-      - id: len_dir_name
+      - id: length
         type: u1
-      - id: len_ext_attr_rec
+      - id: version
+        contents: [ 0x1 ]
+      - id: flags # ToDo, made it too slow for now
```

Comment:
???

And should flags be an own type of ```b1```s?

Reply:
Done